Type definition for Cypress.minimatch is missing function signature

### missing type definition

We include `Cypress.minimatch` type, but it is missing that it can be called as a function

```
cypress/integration/examples/utilities.spec.js:49:22 - error TS2348: Value of type 'IMinimatchStatic' is not callable. Did you mean to include 'new'?

49     const matching = Cypress.minimatch('/users/1/comments', '/users/*/comments', {
                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
50       matchBase: true,
   ~~~~~~~~~~~~~~~~~~~~~~
51     })
```

--- a/cli/package.json
+++ b/cli/package.json
@@ -41,7 +41,7 @@
     "@types/chai-jquery": "1.1.35",
     "@types/jquery": "3.2.16",
     "@types/lodash": "4.14.87",
-    "@types/minimatch": "3.0.1",
+    "@types/minimatch": "3.0.3",
     "@types/mocha": "2.2.44",
     "@types/sinon": "4.0.0",
     "@types/sinon-chai": "2.7.29",

--- a/cli/types/index.d.ts
+++ b/cli/types/index.d.ts
@@ -289,12 +289,18 @@ declare namespace Cypress {
 
     /**
      * Clear data in local storage.
-     * Cypress automatically runs this command before each test to prevent state from being shared across tests. You shouldn’t need to use this command unless you’re using it to clear localStorage inside a single test.
+     * Cypress automatically runs this command before each test to prevent state from being
+     * shared across tests. You shouldn’t need to use this command unless you’re using it
+     * to clear localStorage inside a single test. Yields `localStorage` object.
      *
      * @see https://on.cypress.io/clearlocalstorage
+     * @example
+     *    cy.clearLocalStorage().should(ls => {
+     *      expect(ls.getItem('prop1')).to.be.null
+     *    })
      */
-    clearLocalStorage(key?: string): Chainable<null>
-    clearLocalStorage(re: RegExp): Chainable<null>
+    clearLocalStorage(key?: string): Chainable<Storage>
+    clearLocalStorage(re: RegExp): Chainable<Storage>
 
     /**
      * Click a DOM element.

--- a/cli/types/minimatch.d.ts
+++ b/cli/types/minimatch.d.ts
@@ -3,10 +3,9 @@
 // so that Cypress can get and use the Blob type
 // tslint:disable-next-line:no-implicit-dependencies
 import * as mimimatch from 'minimatch'
-
 export = Mimimatch
 export as namespace Mimimatch
 
 declare namespace Mimimatch {
-  type MimimatchStatic = mimimatch.IMinimatchStatic
+  type MimimatchStatic = mimimatch.IMinimatchStatic & typeof mimimatch
 }

--- a/cli/types/tests/kitchen-sink.ts
+++ b/cli/types/tests/kitchen-sink.ts
@@ -1533,3 +1533,7 @@ cy.wrap('foo').then(subject => {
 }).then(subject => {
   subject // $ExpectType string
 })
+
+Cypress.minimatch('/users/1/comments', '/users/*/comments', {
+  matchBase: true,
+})

--- a/cli/types/tests/kitchen-sink.ts
+++ b/cli/types/tests/kitchen-sink.ts
@@ -1116,6 +1116,14 @@ describe('Kitchen Sink', function() {
       })
 
       // clearLocalStorage() yields the localStorage object
+      cy.clearLocalStorage().should((ls) => {
+        expect(ls.getItem('prop1')).to.be.null
+        expect(ls.getItem('prop2')).to.be.null
+        expect(ls.getItem('prop3')).to.be.null
+        ls.setItem('prop1', 'foo')
+      })
+
+      // clearLocalStorage() yields the localStorage object
       cy.clearLocalStorage()
       cy.window().its('localStorage').should(function(ls) {
         expect(ls.getItem('prop1')).to.be.null

